centralized-sampling-tests/integration-tests/src/main/java/com.amazon.tests/CentralizedSamplingIntegrationTests.java

```diff
@@ -0,0 +1,313 @@
+package com.amazon.tests;
+
+import java.io.*;
+import java.lang.Math;
+import java.util.concurrent.TimeUnit;
+import okhttp3.*;
+import org.springframework.context.annotation.Bean;
+
+/** File that runs all of the integrations tests and creates/deletes sample rules */
+public class CentralizedSamplingIntegrationTests {
+
+  /**
+   * Set up new httpClient
+   *
+   * @return OkhttpClient
+   */
+  @Bean
+  public static Call.Factory httpClient() {
+    return new OkHttpClient();
+  }
+
+  /**
+   * Main function that runs the three tests for Centralized Sampling SampleRulesTests - tests rules
+   * filter and sample the correct number of targets ReservoirTests - tests that reservoir works
+   * correctly with Sampling Rules PriorityTests - test Priority of sampling rules that higher
+   * priority rules get sampled first
+   *
+   * @param args
+   * @throws IOException - throws if calls to xray backend fail
+   * @throws InterruptedException - throws if tests fail
+   */
+  public static void main(String[] args) throws IOException, InterruptedException {
+    sampleRulesTests();
+    reservoirTests();
+    priorityTests();
+  }
+
+  /**
+   * Function to
+   *
+   * @param testCase - testCase to make call with, determines headers, method, and endpoint, etc.
+   * @param sampleRule - sampleRule that is currently active
+   * @return true if call is successful and has expected sampling rate, false else
+   */
+  public static boolean makeCalls(testCases testCase, SampleRule sampleRule) throws IOException {
+    RequestBody reqbody = null;
+    String stringResp = "";
+    if (testCase.method.equals("POST")) {
+      reqbody = RequestBody.create(null, new byte[0]);
+    }
+    try (Response response =
+        httpClient()
+            .newCall(
+                new Request.Builder()
+                    .addHeader("user", testCase.user)
+                    .addHeader("service_name", testCase.name)
+                    .addHeader("required", testCase.required)
+                    .addHeader("totalSpans", String.valueOf(GenericConstants.TOTAL_CALLS))
+                    .url("http://localhost:8080" + testCase.endpoint)
+                    .method(testCase.method, reqbody)
+                    .build())
+            .execute()) {
+      stringResp = response.body().string();
+
+    } catch (IOException e) {
+      throw new UncheckedIOException("Could not fetch endpoint", e);
+    }
+    int expectedRate = GenericConstants.DEFAULT_RATE;
+    if (testCase.matches.contains(sampleRule.getName())) {
+      expectedRate = (int) Math.round(sampleRule.getExpectedSampled() * GenericConstants.TOTAL_CALLS);
+    }
+    double range = expectedRate * .1 + GenericConstants.DEFAULT_RANGE;
+    int roundedRange = (int) Math.round(range);
+    if (expectedRate == 0) {
+      roundedRange = 0;
+    }
+    System.out.println(
```

Comment:
usually use slf4j

Reply:
I'm assuming you mean I should use a logger instead of sout